### PR TITLE
Remove sqlite on memory except tests

### PR DESCRIPTION
**Pull request by sandhilt**
This pull request adds a new feature to generate the SQLite file on demand. Previously, a default SQLite file was used, but now the code has been updated to allow the user to specify a custom SQLite file path. If no path is provided, a temporary directory is created and a new SQLite file is generated in that directory. This provides more flexibility and control over the SQLite file used by the application.

### CODE REVIEW

File: internal/nonodo/nonodo.go [hunk 1]
```diff
@@ -153,7 +154,17 @@ func NewSupervisorHLGraphQL(opts NonodoOpts) supervisor.SupervisorWorker {
 		db = sqlx.MustConnect("postgres", connectionString)
 	} else {
 		slog.Info("Using SQLite ...")
-		db = sqlx.MustConnect("sqlite3", opts.SqliteFile)
+		sqliteFile := opts.SqliteFile
```
Comment (DiegoSena) [inline internal/nonodo/nonodo.go:157]: I think you can move this code to a function that returns db and reuse it below, what do you think ?

**Review by DiegoSena — CHANGES_REQUESTED** [review]
See comment on nonodo file

File: internal/nonodo/nonodo.go (hunk 1 above)
Comment (sandhilt) [reply internal/nonodo/nonodo.go:157]: For sure, I'll change it.

File: internal/nonodo/nonodo.go [hunk 2]
```diff
@@ -255,6 +255,21 @@ func NewSupervisorHLGraphQL(opts NonodoOpts) supervisor.SupervisorWorker {
 	return w
 }
 
+func handleSQLite(opts NonodoOpts) *sqlx.DB {
```
Comment (sandhilt) [inline internal/nonodo/nonodo.go:258]: Check again @DiegoSena

**Review by DiegoSena — APPROVED** [review]

**Pull request merged**